pkg/clients/rds/rds_test.go

```diff
@@ -162,3 +166,689 @@ func TestIsUpToDate(t *testing.T) {
+				Endpoint:                              v1beta1.Endpoint{Port: int(port), HostedZoneID: zone, Address: address},
+				EnhancedMonitoringResourceArn:         arn,
+				LatestRestorableTime:                  &metav1.Time{Time: lastRestoreTime},
+				OptionGroupMemberships:                []v1beta1.OptionGroupMembership{{OptionGroupName: name, Status: status}},
+				PerformanceInsightsEnabled:            insightsEnabled,
+				ReadReplicaDBClusterIdentifiers:       replicaClusters,
+				ReadReplicaDBInstanceIdentifiers:      replicaClusters,
+				ReadReplicaSourceDBInstanceIdentifier: replicaSourceIdentifier,
+				SecondaryAvailabilityZone:             secondaryAZ,
+				StatusInfos: []v1beta1.DBInstanceStatusInfo{{
+					Message:    status,
+					Status:     status,
+					StatusType: status,
+					Normal:     normal,
+				}},
+				VPCSecurityGroups: []v1beta1.VPCSecurityGroupMembership{{
+					Status:             status,
+					VPCSecurityGroupID: name,
+				}},
+			},
+		},
+		"EmptyInstance": {
+			rds:  rds.DBInstance{},
+			want: v1beta1.RDSInstanceObservation{},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			got := GenerateObservation(tc.rds)
+			if diff := cmp.Diff(tc.want, got); diff != "" {
+				t.Errorf("r: -want, +got:\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestGetConnectionDetails(t *testing.T) {
+	address := "testAddress"
+	port := 123
+	cases := map[string]struct {
+		rds  v1beta1.RDSInstance
+		want managed.ConnectionDetails
+	}{
+		"ValidInstance": {
+			rds: v1beta1.RDSInstance{
+				Status: v1beta1.RDSInstanceStatus{
+					AtProvider: v1beta1.RDSInstanceObservation{
+						Endpoint: v1beta1.Endpoint{
+							Address: address,
+							Port:    port,
+						},
+					},
+				},
+			},
+			want: managed.ConnectionDetails{
+				v1alpha1.ResourceCredentialsSecretEndpointKey: []byte(address),
+				v1alpha1.ResourceCredentialsSecretPortKey:     []byte(strconv.Itoa(port)),
+			},
+		},
+		"NilInstance": {
+			rds:  v1beta1.RDSInstance{},
+			want: nil,
+		}}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			got := GetConnectionDetails(tc.rds)
+			if diff := cmp.Diff(tc.want, got); diff != "" {
+				t.Errorf("r: -want, +got:\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestLateInitialize(t *testing.T) {
+	name := "testName"
+	clusterName := "testCluster"
+	value := "testValue"
+	description := "testDescription"
+	status := "testStatus"
+	arn := "my:arn"
+	zone := "zone"
+	az := "az"
+	vpc := "vpc"
+	port := 123
+	port64 := int64(port)
+	storage := 1
+	storage64 := int64(storage)
+	retention := 2
+	retention64 := int64(retention)
+	monitoring := 3
+	monitoring64 := int64(monitoring)
+	tier := 4
+	tier64 := int64(tier)
+	storageType := "storageType"
+	instanceClass := "class"
+	engine := "5.6.41"
+	truncEngine := "5.6"
+	multiAZ := false
+	trueFlag := true
+	kmsID := "kms"
+	username := "username"
+	window := "window"
+	existingName := "existing"
```

Comment:
nitpick: looks like these values are used across a number of the tests, how would you feel about moving them out to a global `var` block so they do not have to be reproduced in each?

Reply:
Works for me. Will get that up when I can

Reply:
Sounds good :) Ping me when ready for another review :+1: